src/iccread.rs

```diff
@@ -954,6 +985,467 @@ pub extern "C" fn qcms_white_point_sRGB() -> qcms_CIE_xyY {
+                //
+                // Lc = 10^(2.5*V - 2.5)  for 1 >= V >= 0
+                let table = build_trc_table(NUM_TRC_TABLE_ENTRIES, |v| 10f64.powf(2.5 * v - 2.5));
+                curveType::Curve(table)
+            }
+            TransferCharacteristics::Iec61966 => unimplemented!(),
+            TransferCharacteristics::Bt_1361 => unimplemented!(),
+            TransferCharacteristics::Srgb => {
+                // Should we prefer this or curveType::Parametric?
+                curveType::Curve(build_sRGB_gamma_table(NUM_TRC_TABLE_ENTRIES))
+            }
+
+            TransferCharacteristics::Smpte2084 => {
+                // Despite using Lo rather than Lc, H.273 gives the OETF:
+                //
+                // V = ( ( c1 + c2 * (Lo)^n ) ÷ ( 1 + c3 * (Lo)^n ) )^m
+                const c1: f64 = 0.8359375;
+                const c2: f64 = 18.8515625;
+                const c3: f64 = 18.6875;
+                const m: f64 = 78.84375;
+                const n: f64 = 0.1593017578125;
+
+                // Inverting this to give the EOTF required for the profile
+                // (and confirmed by Rec. ITU-R BT.2100-2, Table 4) gives
+                //
+                // Y = ( max[( X^(1/m) - c1 ), 0] ÷ ( c2 - c3 * X^(1/m) ) )^(1/n)
+                let table = build_trc_table(NUM_TRC_TABLE_ENTRIES, |x| {
+                    ((x.powf(1. / m) - c1).max(0.) / (c2 - c3 * x.powf(1. / m))).powf(1. / n)
+                });
+                curveType::Curve(table)
+            }
+            TransferCharacteristics::Smpte428 => unimplemented!(),
+            TransferCharacteristics::Hlg => {
+                // The opto-electronic transfer characteristic function (OETF)
+                // as defined in ITU-T H.273 table 3, row 18:
+                //
+                // V = a * Ln(12 * Lc - b) + c  for 1      >= Lc >  1 ÷ 12
+                // V = Sqrt(3) * Lc^0.5         for 1 ÷ 12 >= Lc >= 0
+                const a: f64 = 0.17883277;
+                const b: f64 = 0.28466892;
+                const c: f64 = 0.55991073;
+
+                // Inverting this to give the EOTF required for the profile
+                // (and confirmed by Rec. ITU-R BT.2100-2, Table 4) gives
+                //
+                // Y = (X^2) / 3             for 0   <= X <= 0.5
+                // Y = ((e^((X-c)/a))+b)/12  for 0.5 <  X <= 1
+                let table = build_trc_table(NUM_TRC_TABLE_ENTRIES, |x| {
+                    if x <= 0.5 {
+                        let y1 = x.powf(2.) / 3.;
+                        assert!(y1 >= 0. && y1 <= 1. / 12.);
+                        y1
+                    } else {
+                        let y2 = (std::f64::consts::E.powf((x - c) / a) + b) / 12.;
+                        assert!(y2 > 1. / 12. && y2 <= 1.);
+                        y2
+                    }
+                });
+                curveType::Curve(table)
+            }
+        }
+    }
+}
+
+#[cfg(feature = "c_bindings")]
```

Comment:
Because there's no pure rust alternative for `qcms_profile_from_path`. It wouldn't be that hard to write one, but nor did there seem to be much benefit over just using the existing code.

If you think it'd be valuable for non-test code, I could look into implementing that. It always seems beneficial to reduce the surface area of unsafe code.

Reply:
There's this https://github.com/FirefoxGraphics/qcms/blob/c0ac055403f9b0f7a2bdaec61561338a70cab7b9/src/gtest.rs#L764

Reply:
Actually, this is the one you'd want: https://github.com/FirefoxGraphics/qcms/blob/c0ac055403f9b0f7a2bdaec61561338a70cab7b9/src/gtest.rs#L907